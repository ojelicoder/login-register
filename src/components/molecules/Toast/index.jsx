import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { t } from "i18next";

const Toast = (props) => {
  const [toastShow, setToastShow] = useState(true),
    { message, type, toastClosed } = props;

  useEffect(() => {
    setToastShow(false);

    setTimeout(() => {
      setToastShow(false);
      toastClosed();
    }, 2000);
  }, [toastClosed]);

  return (
    <div className={toastShow ? "toast" : "toast in"}>
      <div className={type === "success" ? "toast-success" : "toast-error"}>
        <div className="toast_content">{t(message)}</div>
      </div>
    </div>
  );
};

Toast.propTypes = {
  message: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  toastClosed: PropTypes.func,
};

export default Toast;
