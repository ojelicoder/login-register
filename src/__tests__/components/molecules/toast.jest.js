import React from "react";
import Toast from "../../../components/molecules/Toast";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";

describe("Toast", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<Toast />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Renders with a className equal to the props", () => {
    const toastProps = {
      type: "success",
      message: "general.message.toaster.login.success",
    };
    const { container } = render(<Toast {...toastProps} />);
    expect(container.getElementsByClassName("toast-success").length).toBe(1);
  });
});
