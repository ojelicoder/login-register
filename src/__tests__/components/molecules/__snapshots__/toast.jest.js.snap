// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders correctly 1`] = `
<div
  className="toast"
>
  <div
    className="toast-error"
  >
    <div
      className="toast_content"
    />
  </div>
</div>
`;
